docker/web/Dockerfile.release

```diff
@@ -0,0 +1,110 @@
+# Define ARG at the top level
+ARG NODE_VERSION=10
+ARG WEB_BASE_HREF=/
+ARG NGINX_RUNTIME_SRC=./nginx-runtime
+ARG VUE_ON_NGINX_SRC=./vue-on-nginx
+ARG WEB_SRC=./web
+ARG NPM_INSTALL_ARGS=""
+
+###################################################################################
+# 1. Build web-runtime
+###################################################################################
+# Use the offical nginx (based on debian)
+FROM nginx:stable AS runtime
+ARG NGINX_RUNTIME_SRC
+ARG VUE_ON_NGINX_SRC
+
+ENV STI_SCRIPTS_PATH=/usr/libexec/s2i
+
+# Required for HTTP Basic feature
+RUN apt-get update -y && \
+    apt-get install -y openssl ca-certificates && \
+    rm -rf /var/lib/apt/lists/*
+
+# Copy our OpenShift s2i scripts over to default location
+COPY ${VUE_ON_NGINX_SRC}/s2i/bin/ /usr/libexec/s2i/
+
+# Expose this variable to OpenShift
+LABEL io.openshift.s2i.scripts-url=image:///usr/libexec/s2i
+
+# Copy config from source to container
+COPY ${NGINX_RUNTIME_SRC}/nginx.conf.template /tmp/
+
+# Copy run script
+COPY ${NGINX_RUNTIME_SRC}/s2i/bin/run /usr/libexec/s2i/run
+
+# =================================================================================
+# Fix up permissions
+# ref: https://torstenwalter.de/openshift/nginx/2017/08/04/nginx-on-openshift.html
+# - S2I scripts must be executable
+# - Make sure nginx can read and write it's working directories.
+# - The container dynamically configures nginx on startup
+# - The application artifacts live in /tmp
+# ---------------------------------------------------------------------------------
+RUN chmod -R g+rwx $STI_SCRIPTS_PATH
+RUN chmod g+rw /var/cache/nginx \ 
+    /var/run \
+    /var/log/nginx \
+    /etc/nginx/nginx.conf \
+    /tmp
+
+# =================================================================================
+
+# Work-around for issues with S2I builds on Windows
+WORKDIR /tmp
+
+# Nginx runs on port 8080 by default
+EXPOSE 8080
+
+# Switch to usermode
+USER 104
+
+###################################################################################
+# 2. Build web-artifacts
+###################################################################################
+# Update to later nodejs version when application is upgrade to Vue 2.7 or later. The centos version has python2 installed which is required for the current application version.
+# FROM registry.access.redhat.com/ubi9/nodejs-${NODE_VERSION} AS artifacts 
+FROM registry.hub.docker.com/centos/nodejs-12-centos7 AS artifacts
```

Comment:
I see a ToDo here.  Has this been captured in a ticket?

Reply:
https://jag.gov.bc.ca/jira/browse/SCV-501